File: Lab3_back_end/src/bl/BaseBL.java
package bl;

import java.util.List;

/**
 *
 * @author _Adrián_Prendas_
 */
public interface BaseBL<T,K> {
    //CRUD
    public boolean create(T o);
    public T read(K key);
    public List<T> read();
    public boolean update(T o);
    public boolean delete(K key);
}

File: 03/app/src/main/java/com/example/bl/BaseBL.kt
package com.example.bl

/**
 * Created by Josue on 25/03/2018.
 */
interface BaseBl<K,T>{
    fun create(obj: T): T
    fun read(key: K): T?
    fun readAll(): List<T>
    fun update(obj: T): T?
    fun delete(key: K): T?
}